indra/llprimitive/tests/llgltfmaterial_test.cpp

```diff
@@ -0,0 +1,256 @@
+        for (LLGLTFMaterial::TextureInfo i = LLGLTFMaterial::GLTF_TEXTURE_INFO_BASE_COLOR; i < LLGLTFMaterial::GLTF_TEXTURE_INFO_COUNT; i = LLGLTFMaterial::TextureInfo((U32)i + 1))
+        {
+            material.setTextureOffset(i, test_transform.mOffset);
+            material.setTextureScale(i, test_transform.mScale);
+            material.setTextureRotation(i, test_transform.mRotation);
+        }
+    }
+
+    void apply_test_material_factors(LLGLTFMaterial& material)
+    {
+        material.setBaseColorFactor(LLColor4(test_fraction_big, test_fraction_big, test_fraction_big, test_fraction_big));
+        material.setEmissiveColorFactor(LLColor3(test_fraction_big, test_fraction_big, test_fraction_big));
+        material.setMetallicFactor(test_fraction);
+        material.setRoughnessFactor(test_fraction);
+    }
+
+    LLGLTFMaterial create_test_material()
+    {
+        LLGLTFMaterial material;
+
+        apply_test_material_texture_ids(material);
+
+        apply_test_material_texture_transforms(material);
+
+        apply_test_material_factors(material);
+
+        material.setAlphaCutoff(test_fraction);
+        // Because this is the default value, it should append to the extras field to mark it as an override
+        material.setAlphaMode(LLGLTFMaterial::ALPHA_MODE_OPAQUE);
+        // Because this is the default value, it should append to the extras field to mark it as an override
+        material.setDoubleSided(false);
+
+        return material;
+    }
+
+    void ensure_gltf_material_serialize(const std::string& ensure_suffix, const LLGLTFMaterial& material_in)
+    {
+        const std::string json_in = material_in.asJSON();
+        LLGLTFMaterial material_out;
+        std::string warn_msg;
+        std::string error_msg;
+        bool serialize_success = material_out.fromJSON(json_in, warn_msg, error_msg);
+        ensure_equals("LLGLTFMaterial serialization has no warnings: " + ensure_suffix, "", warn_msg);
+        ensure_equals("LLGLTFMaterial serialization has no errors: " + ensure_suffix, "", error_msg);
+        ensure("LLGLTFMaterial serializes successfully: " + ensure_suffix, serialize_success);
+        ensure("LLGLTFMaterial is preserved when deserialized: " + ensure_suffix, material_in == material_out);
+        const std::string json_out = material_out.asJSON();
+        ensure_equals("LLGLTFMaterial is preserved when serialized: " + ensure_suffix, json_in, json_out);
+    }
+
+    void ensure_gltf_material_trimmed(const std::string& material_json, const std::string& must_not_contain)
+    {
+        ensure("LLGLTFMaterial serialization trims property '" + must_not_contain + "'", material_json.find(must_not_contain) == std::string::npos);
+    }
+
+    // Test that GLTF material fields have not changed since these tests were written
+    template<> template<>
+    void llgltfmaterial_object_t::test<1>()
+    {
+        if (sizeof(void*) > 4) // Don't bother running this test for 32-bit systems
```

Comment:
we have a preprocessor definition ADDRESS_SIZE for this kind of thing.  see `indra/cmake/00-Common.cmake:30` for where it is set